addons/product/models/product_attribute.py

```diff
@@ -78,6 +81,12 @@ def _compute_products(self):
     def _without_no_variant_attributes(self):
         return self.filtered(lambda pa: pa.create_variant != 'no_variant')
 
+    @api.depends('display_type', 'number_related_products')
+    def _compute_create_variant(self):
+        for record in self:
+            if record.display_type == "multi" and record.number_related_products == 0:
+                record.create_variant = "no_variant"
```

Comment:
```suggestion
            if record.display_type == 'multi' and record.number_related_products == 0:
                record.create_variant = 'no_variant'
```

Reply:
Also move this method [Here](https://github.com/odoo/odoo/pull/160270/files#diff-61d64e451677b69b389f615bcafde2b204c5d5a79a2133dbdadf10d8f3c42774R70)
This way compute method sequence should be same as field defined